strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetPackageController.java

```diff
@@ -154,19 +179,100 @@ public ResponseEntity deletePackage(@RequestHeader(name = "X-NuGet-ApiKey", requ
 
         return new ResponseEntity<>(HttpStatus.OK);
     }
     
+    private PackageFeed transform(String feedId,
+                                  Collection<? extends Nupkg> files,
+                                  String orderBy,
+                                  Integer skip,
+                                  Integer top)
+    {
+        PackageFeed feed = new PackageFeed();
+        // feed.setId(getContext().getRootUri().toString());
+        feed.setId(feedId);
+        feed.setUpdated(new Date());
+        feed.setTitle("Packages");
+        List<PackageEntry> packageEntrys = new ArrayList<>();
+        for (Nupkg nupkg : files)
+        {
+            try
+            {
+                PackageEntry entry = createPackageEntry(feedId, (PathNupkg) nupkg);
+                calculateFeedEntryProperties((PathNupkg) nupkg, entry.getProperties());
+                packageEntrys.add(entry);
+            }
+            catch (NoSuchAlgorithmException | IOException | NugetFormatException e)
+            {
+                logger.error("Failed to parse package " + nupkg, e);
+            }
+        }
+        Collections.sort(packageEntrys, getPackageComparator(orderBy));
+        logger.debug("Got {} packages", new Object[] { packageEntrys.size() });
+        //TODO: SB-1004
+        packageEntrys = packageEntrys.stream().skip(skip).limit(top).collect(Collectors.toList());
+        logger.debug("Prepared {} packages", new Object[] { packageEntrys.size() });
+        feed.setEntries(packageEntrys);
+        return feed;
+    }
+
+    private void calculateFeedEntryProperties(PathNupkg nupkg,
+                                              EntryProperties properties)
+    {
+        RepositoryPath path = nupkg.getPath();
+        ArtifactEntry artifactEntry = path.getArtifactEntry();
+
+        properties.setId(nupkg.getId());
+        
+        properties.setReportAbuseUrl("");
+
+        properties.setDownloadCount(artifactEntry.getDownloadCount());
+        properties.setVersionDownloadCount(artifactEntry.getDownloadCount());
+
+        properties.setRatingsCount(0);
+        properties.setVersionRatingsCount(0);
+
+        properties.setRating(Double.valueOf(0));
+        properties.setVersionRating(Double.valueOf(0));
+
+        ArtifactTag lastVersionTag = artifactTagService.findOneOrCreate(ArtifactTagEntry.LAST_VERSION);
+        properties.setIsLatestVersion(artifactEntry.getTagSet().contains(lastVersionTag));
+    }
+
+    private PackageEntry createPackageEntry(String feedId,
+                                            PathNupkg nupkg)
+        throws NoSuchAlgorithmException,
+        IOException,
+        NugetFormatException
+    {
+        return new PackageEntry(nupkg){
+
+            @Override
+            protected String getRootUri()
+            {
+                return feedId;
+            }
+        };
+    }
+
+    protected Comparator<PackageEntry> getPackageComparator(final String orderByClause) {
```

Comment:
Cosmetic: The indentation of this method seems incorrect.

Reply:
fixed